python/hail/contrib/ldsc.py

```diff
@@ -0,0 +1,156 @@
+
+import numpy as np
+import hail as hl
+from hail.table import Table
+from hail.linalg import BlockMatrix
+from hail.typecheck import *
+from hail.expr.expressions import *
+from hail.utils import new_temp_file, wrap_to_list
+
+@typecheck(entry_expr=expr_numeric,
+           annotation_exprs=oneof(expr_numeric, sequenceof(expr_numeric)),
+           position_expr=expr_numeric,
+           window_size=oneof(int, float))
+def ld_score(entry_expr, annotation_exprs, position_expr, window_size) -> Table:
+    """
+    Calculate LD scores.
+
+    Example
+    -------
+
+    >>> # Load genetic data into MatrixTable
+    >>> mt = hl.import_plink(bed='data/ldsc.bed', 
+    ...                      bim='data/ldsc.bim', 
+    ...                      fam='data/ldsc.fam')
+    
+    >>> # Create locus-keyed Table with numeric variant annotations
+    >>> ht = hl.import_table('data/ldsc.annot', types={'BP': hl.tint, 'binary': hl.tfloat, 'continuous': hl.tfloat})
+    >>> ht = ht.annotate(locus=hl.locus(ht.CHR, ht.BP))  
+    >>> ht = ht.key_by('locus')
+    
+    >>> # Annotate MatrixTable with external annotations 
+    >>> mt = mt.annotate_rows(univariate_annotation=hl.literal(1),   # univariate LD scores
+    ...                       binary_annotation=ht[mt.locus].binary, 
+    ...                       continuous_annotation=ht[mt.locus].continuous)
+    ...
+    >>> # Annotate MatrixTable with alt allele count stats
+    >>> mt = mt.annotate_rows(stats=hl.agg.stats(mt.GT.n_alt_alleles()))
+     
+    >>> # Calculate LD score for each variant, annotation using standardized genotypes
+    >>> ht_scores = hl.ld_score(entry_expr=hl.or_else((mt.GT.n_alt_alleles() - mt.stats.mean)/mt.stats.stdev, 0.0),
+    ...                         annotation_exprs=[mt.univariate_annotation, mt.binary_annotation, mt.continuous_annotation],
+    ...                         position_expr=mt.cm_position,
+    ...                         window_size=1)
+
+    Warning
+    -------
+    :func:`.ld_score` will fail if ``entry_expr`` results in any missing values. The special float value ``nan`` is not considered a missing value.
+
+
+    **Further reading**
+
+    For more in-depth discussion of LD scores, see:
+
+    - `LD Score regression distinguishes confounding from polygenicity in genome-wide association studies (Bulik-Sullivan et al, 2015) <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/>`__
```

Comment:
in our main codebase, we try to keep lines a bit shorter. I wrote a tool to automate this by block:

https://hail.is/tools/formatter.html

Reply:
I'm not sure how to split back-ticked phrases across multiple lines.

Reply:
I think it'll work as expected, the link will stay. We can check by looking at the docs when they build

Reply:
I'm getting build errors trying to split the lines. Happy to change this later if we figure out how.

Reply:
ok, dont worry about it